02-peer-protocol.md

```diff
@@ -1713,6 +1784,179 @@ satoshis, which is possible if `dust_limit_satoshis` is below 546 satoshis).
+    - SHOULD ignore `closing_complete`.
+    - SHOULD send a `warning`.
+    - SHOULD close the connection.
+  - MUST generate the remote closing transaction as specified in [BOLT #3](03-transactions.md#closing-transaction).
+  - Select a signature for validation:
+    - If the local output amount is dust:
+      - MUST use `closer_output_only`.
+    - Otherwise, if it considers the local output amount uneconomical AND its `closee_scriptpubkey` is not `OP_RETURN`:
+      - MUST use `closer_output_only`.
+    - Otherwise, if `closer_and_closee_outputs` is present:
+      - MUST use `closer_and_closee_outputs`.
+    - Otherwise:
+      - MUST use `closee_output_only`.
+  - If the selected signature field does not exist:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is not valid for the corresponding closing transaction specified in [BOLT #3](03-transactions.md#closing-transaction):
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is non-compliant with LOW-S-standard rule<sup>[LOWS](https://github.com/bitcoin/bitcoin/pull/6769)</sup>:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - MUST sign and broadcast the corresponding closing transaction.
+  - MUST send `closing_sig` with a single valid signature in the same TLV field as the `closing_complete`.
+  - MUST use `closer_scriptpubkey` for its own future `closing_complete` messages.
+
+The receiver of `closing_sig`:
+  - If `closer_scriptpubkey`, `closee_scriptpubkey`, `fee_satoshis` or `locktime` don't match what was sent in `closing_complete`:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If `tlvs` does not contain exactly one signature:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If `tlvs` does not contain one of the TLV fields sent in `closing_complete`:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is not valid for the corresponding closing transaction specified in [BOLT #3](03-transactions.md#closing-transaction):
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - If the signature field is non-compliant with LOW-S-standard rule<sup>[LOWS](https://github.com/bitcoin/bitcoin/pull/6769)</sup>:
+    - MUST either send a `warning` and close the connection, or send an `error` and fail the channel.
+  - otherwise:
+    - MUST broadcast the corresponding closing transaction.
+  - MAY send another `closing_complete` (e.g. with a different `fee_satoshis` or `closer_scriptpubkey`).
+
+### Rationale
```

Comment:
Can we add rationale here for why both of the new messages carry the closing scripts for both parties?

Reply:
One thing that isn't immediately clear is: why does `closing_singed` need to replicate these fields? As a peer you MUST remember the last address you sent to make sure they don't trick you into sending all your funds away. Therefore, isn't it redundant to send the scripts _again_ for the response message?